.clang-format

```diff
@@ -0,0 +1,219 @@
+LambdaBodyIndentation: Signature
+
+# No indent in namespaces.
+NamespaceIndentation: None
+
+# No way to specify C vs C++ style comment for namespace closure, so just
+# leave them alone.
+FixNamespaceComments: false
+
+# 1 space before trailing comments
+SpacesBeforeTrailingComments: 1
+
+# Colon spacing should always be '"key" : "value"'
+BitFieldColonSpacing: Both
+
+# Do not break up strings into multi-line strings. This becomes a problem when
+# multi-line strings with no spaces are broken, which upsets the Tup code checker.
+BreakStringLiterals: false
+
+# C-tor should always break before the : and again before the {} in cases where
+# otherwise the c-tor could theoretically be one-lined. In the case where the
+# initializer list is short enough to fit on one line after the wrap, do not break
+# into individual lines.
+# Note NextLineOnly requires clang-format >= 15
+PackConstructorInitializers: NextLineOnly
+
+# includes are sorted based on the other suboptions below
+SortIncludes:    true
+# Merge multiple #include blocks together and sort as one. Then split into groups based on
+# category priority.
+IncludeBlocks: Regroup
+# Regular expressions denoting the different #include categories used for ordering #includes.
+IncludeCategories:
+  - Regex:      'debug.h'
+    Priority:   4
+  - Regex:      '\"\./[^/]+\"'
+    Priority:   1
+  - Regex:      '\"[^/]+\"'
+    Priority:   8
+  - Regex:	'boost'
+    Priority:   5
+  - Regex:      '<.*\.h'
+    Priority:   2
+  - Regex:      '<'
+    Priority:   3
+  - Regex:      '.*'
+    Priority:   6
+
+# use "template <>" instead of "template<>"
+SpaceAfterTemplateKeyword: true
+
+# Always pad assignments with a space, like "variable = 2;".
+SpaceBeforeAssignmentOperators: true
+
+# Turn off comment reflow entirely
+# TODO: This was done to prevent undoing good formatting of doxygen @brief @param
+#       etc multi-line comments (similar to this comment). Investigate comment
+#       pragmas to leave these alone?
+ReflowComments: false
+
+# There should always be an empty line between method definiton blocks.
+SeparateDefinitionBlocks: Always
+
+# The penalty for each line break introduced inside a comment.
+PenaltyBreakComment: 0
+
+# How many spaces are allowed at the start of a line comment. To disable the maximum
+# set it to -1, apart from that the maximum takes precedence over the minimum
+SpacesInLineCommentPrefix:
+  Minimum: 1
+  Maximum: -1
+
+---
+
+Language: Proto
+
+# Google style is a good starting point;
+BasedOnStyle: Google
+IndentWidth: 2
+
+# The intended impact here is to allow up to 96 characters, but break at 80 if
+# 96 is exceeded.  The "Penalty" concept in clang-format is not well
+# documented, see here for more: https://stackoverflow.com/questions/26635370
+ColumnLimit: 80
+PenaltyExcessCharacter: 50
+
+# Turn off comment reflow entirely
+# TODO: This was done to prevent undoing user-defined formatting of multi-line
+#       comments.
```

Comment:
... and because clang-format is known (?) to be inconsistent in re-flowing from one run to another